### PR TITLE
add search functionality

### PR DESCRIPTION
**Pull request by rajneeshhp**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,59 +1,67 @@
...
-    this.setState({ data: [...this.state.data, newVal] })
+  const serText = (serVal) => {
+    setFilterText(serVal)
+  }
 
+  const handleSubmit = (newVal, todo) => {
+    const arrData = data.map((item) => {
+      return item.todo
+    })
+    const xyx = arrData.map(e => e.toLocaleLowerCase()).includes(todo.toLocaleLowerCase());
```
Comment (dev22451) [inline src/App.js]: what we are doing here... ?

Comment (dev22451) [reply src/App.js]: do proper naming for variables and functions .. don't use xyx or abc

File: src/App.js [hunk 2]
```diff
@@ -1,59 +1,67 @@
...
-  }
+const App = (props) => {
 
+  const [data, setData] = useState([{ id: new Date().getTime().toString(), todo: "Apple" }]);
+  const [filterText, setFilterText] = useState('');
 
-  handleSubmit = (newVal) => {
 
-    this.setState({ data: [...this.state.data, newVal] })
+  const serText = (serVal) => {
```
Comment (dev22451) [inline src/App.js:22]: we can write without curly braces if there is only one statement in that function
like 
const serText = (serVal) => setFilterText(serVal)

File: src/App.js [hunk 3]
```diff
@@ -1,59 +1,67 @@
...
+    const updateItems = data.filter((item) => {
+      return index !== item.id;
     })
+    setData(updateItems);
   }
 
-  handleOnEdit = (editVal, id) => {
-    const { data } = this.state
+  const handleOnEdit = (editVal, id) => {
+    const abc = [...data]
```
Comment (dev22451) [inline src/App.js]: do proper naming for variable

File: src/component/listitem.js [hunk 4]
```diff
@@ -1,102 +1,94 @@
...
-class ListItem extends React.Component {
-    //myRef = React.createRef();
-    state = {
-        onEdit: false,
-        editVal: this.props.item,
-        isChecked: false
+const ListItem = (props) => {
 
-    }
+    const [edit, setonEdit] = useState(false);
```
Comment (dev22451) [inline src/component/listitem.js:9]: setOnEdit will looks good 😄

File: src/App.js (hunk 1 above)
Comment (rajneeshhp) [reply src/App.js]: ok

**Review by VishalKrHangingPanda — APPROVED** [review]

**Pull request merged**